pkg/drivers/pgsql/pgsql.go

```diff
@@ -153,9 +154,10 @@ func createDBIfNotExist(dataSourceName string) error {
 		logrus.Tracef("SETUP EXEC : %v", util.Stripped(stmt))
 		_, err = db.Exec(stmt)
```

Comment:
This leaves us trying to use `db`, but if we don't return above when the open fails, we're operating with nil `db`, which will crash when you try to access the Exec function on a nil pointer.

Reply:
With the return on line 141, this shouldn't be possible anymore.